deerlab/snlls.py

```diff
@@ -422,17 +456,38 @@ def ci(coverage,ptype='full'):
         stats = stats[0]
         fvals = fvals[0]
 
+    for i in range(len(subsets)):
+        scales[i] *= prescales[i]
+
     # Display function
     def plotfcn(show=False):
         fig = _plot(subsets,y,yfit,show)
         return fig
 
-    return FitResult(nonlin=nonlinfit, lin=linfit, uncertainty=paramuq, regparam=alpha, plot=plotfcn,
-                     stats=stats, cost=fvals, residuals=sol.fun, success=sol.success)
+    return FitResult(nonlin=nonlinfit, lin=linfit, nonlinUncert=paramuq_nonlin, linUncert=paramuq_lin, regparam=alpha, plot=plotfcn,
+                     stats=stats, cost=fvals, residuals=sol.fun, success=sol.success, scale=scales)
 # ===========================================================================================
 
 
-def _augment(res, J, regtype, alpha, L, x, eta, Nnonlin):
+def uq_subset(uq_full,subset):
+#===========================================================================
+    "Wrapper around the CI function handle of the uncertainty structure"
+    uq_subset = copy.deepcopy(uq_full)
+
+    uq_subset.mean = uq_subset.mean[subset]
+    uq_subset.median = uq_subset.median[subset]
+    uq_subset.std = uq_subset.std[subset]
+    uq_subset.covmat = uq_subset.covmat[np.ix_(subset,subset)]
+    uq_subset.nparam = len(subset)
+
+    # Get requested confidence interval of joined parameter set
+    uq_subset.ci = lambda coverage: uq_full.ci(coverage)[subset, :]
+    uq_subset.percentile = lambda p: uq_full.percentile(p)[subset]
+
+    return uq_subset
+#===========================================================================
+
+def reg_penalty(regtype, alpha, L, x, eta, Nnonlin):

```

Comment:
Rename to `augment_residual` or similar.

Reply:
`augment_residual ` is no longer an appropriate name as the function no longer augments the residual and Jacobian. Instead it is returning the residual and Jacobian subsets corresponding to the regularization penalty. The augmentation or copying is done later in the main code.